pkg/application/dao/dao.go

```diff
@@ -285,54 +285,49 @@ func (d *dao) TransferByID(ctx context.Context, id uint, groupID uint) error {
 func (d *dao) List(ctx context.Context, groupIDs []uint,
 	query *q.Query) (int, []*models.Application, error) {
 	var (
-		applications []*models.Application
-		total        int64
+		applications   []*models.Application
+		total          int64
+		finalStatement *gorm.DB
 	)
 
-	finalStatement := d.db.WithContext(ctx).Table("tb_application as a")
-	if query != nil {
-		// basic filter
+	genSQL := func() *gorm.DB {
+		statement := d.db.WithContext(ctx).Table("tb_application as a").Select("a.*")
 		for k, v := range query.Keywords {
 			switch k {
 			case corecommon.ApplicationQueryName:
-				finalStatement = finalStatement.Where("a.name like ?", fmt.Sprintf("%%%v%%", v))
+				statement = statement.Where("a.name like ?", fmt.Sprintf("%%%v%%", v))
 			case corecommon.ApplicationQueryByTemplate:
-				finalStatement = finalStatement.Where("a.template = ?", v)
+				statement = statement.Where("a.template = ?", v)
 			case corecommon.ApplicationQueryByRelease:
-				finalStatement = finalStatement.Where("a.template_release = ?", v)
+				statement = statement.Where("a.template_release = ?", v)
 			}
 		}
+		statement = statement.Where("a.deleted_ts = 0")
+		return statement
+	}
 
+	if query != nil {
+		// basic filter
+		finalStatement = genSQL()
 		// add groupID filter
 		if len(groupIDs) != 0 {
 			finalStatement = finalStatement.Where("group_id in ?", groupIDs)
 		}
-		finalStatement = finalStatement.Where("a.deleted_ts = 0")
 
 		// union user direct authorized entity
```

Comment:
// make it more clear, groupIDs  in userid exist and not exist are not the same meaning. there are  mutual exclusion

if (userid exist) { 
  // direct member
  +
  // groupIDs  means the inherit  authorized app 
} else {
 // groupIDs  means filter the applications in those groups
}

Reply:
fixed